opengrok-indexer/src/main/java/org/opengrok/indexer/Metrics.java

```diff
@@ -53,90 +57,134 @@
-        }
+    private GraphiteConfig getGraphiteConfig() {
+        return new GraphiteConfig() {
+            @Override
+            public String get(String k) {
+                return null;
+            }
+
+            @Override
+            public int port() {
+                return RuntimeEnvironment.getInstance().getGraphiteConfig().getPort();
+            }
+
+            @Override
+            public String host() {
+                return RuntimeEnvironment.getInstance().getGraphiteConfig().getHost();
+            }
+
+            @Override
+            public GraphiteProtocol protocol() {
+                return RuntimeEnvironment.getInstance().getGraphiteConfig().getProtocol();
+            }
+        };
+    }
 
-        @Override
-        public int port() {
-            return RuntimeEnvironment.getInstance().getStatsdConfig().getPort();
-        }
+    private static PrometheusMeterRegistry prometheusRegistry;
 
-        @Override
-        public String host() {
-            return RuntimeEnvironment.getInstance().getStatsdConfig().getHost();
-        }
+    private MeterRegistry activeRegistry;
 
-        @Override
-        public boolean buffered() {
-            return true;
-        }
-    };
+    private static final Metrics instance = new Metrics();
 
-    private static PrometheusMeterRegistry prometheusRegistry;
-    private static StatsdMeterRegistry statsdRegistry;
-
-    static {
-        MeterRegistry registry = null;
-
-        if (RuntimeEnvironment.getInstance().getStatsdConfig().isEnabled()) {
-            LOGGER.log(Level.INFO, "configuring StatsdRegistry");
-            statsdRegistry = new StatsdMeterRegistry(statsdConfig, Clock.SYSTEM);
-            registry = statsdRegistry;
-        } else if (!RuntimeEnvironment.getInstance().isIndexer()) {
-            LOGGER.log(Level.INFO, "configuring PrometheusRegistry");
-            prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
-            registry = prometheusRegistry;
-        }
+    private Metrics() {
+    }
 
-        if (registry != null) {
-            new ClassLoaderMetrics().bindTo(registry);
-            new JvmMemoryMetrics().bindTo(registry);
-            new JvmGcMetrics().bindTo(registry);
-            new ProcessorMetrics().bindTo(registry);
-            new JvmThreadMetrics().bindTo(registry);
-        }
+    /**
+     * @return the only instance of Metrics
+     */
+    public static Metrics getInstance() {
+        return instance;
     }
 
-    private Metrics() {
+    /**
+     * Configure meter registry.
+     * @param type type of meter registry
+     */
+    public void configure(Configuration.supportedMeterRegistryType type) {
+        switch (type) {
+            case PROMETHEUS:
+                LOGGER.log(Level.INFO, "configuring Prometheus registry");
+                prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
prometheusRegistry does not need to be static if you set it inside an instance method. You still use this class as a singleton so you'll always have an instance from getInstance()

Reply:
right, good catch. This is leftover from before getInstance().